### PR TITLE
Live tests for Form Recognizer

### PR DESCRIPTION
**Pull request by samvaity**
- Setup live test pipeline for Form Recognizer
- Add test-reosurces.json ARM template for creating and tearing down resources.
- Fix failing tests in record mode.  
- Working live pipeline build - https://dev.azure.com/azure-sdk/internal/_build/results?buildId=363726&view=results

### CODE REVIEW

File: sdk/formrecognizer/tests.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+trigger: none
+
+jobs:
+  - template: ../../eng/pipelines/templates/jobs/archetype-sdk-tests.yml
+    parameters:
+      ServiceDirectory: storage
```
Comment (danieljurek) [inline sdk/formrecognizer/tests.yml:6]: this should be `formrecognizer`

Comment (danieljurek) [inline sdk/formrecognizer/tests.yml:6]:
```suggestion
      ServiceDirectory: formrecognizer
```

File: sdk/formrecognizer/test-resources.json [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+                            "enabled": true
+                        }
+                    },
+                    "keySource": "Microsoft.Storage"
+                },
+                "accessTier": "Hot"
+            }
+        },
+        {
+            "type": "Microsoft.Storage/storageAccounts",
```
Comment (danieljurek) [inline sdk/formrecognizer/test-resources.json:100]: Might be able to get rid of the secondary storage account if it's not needed.

File: sdk/formrecognizer/test-resources.json [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+            "type": "string",
+            "defaultValue": "core.windows.net"
+        },
+        "testApplicationOid": {
+            "type": "string",
+            "metadata": {
+                "description": "The principal to assign the role to. This is application object id."
+            }
+        },
+        "formRecognizerBaseName": {
```
Comment (danieljurek) [inline sdk/formrecognizer/test-resources.json:18]: `formRecognizerBaseName` is redundant, you can just use the `baseName` and and add any additional formatting in a variable.

File: sdk/formrecognizer/azure-ai-formrecognizer/src/test/java/com/azure/ai/formrecognizer/FormTrainingClientTestBase.java [hunk 4]
```diff
@@ -182,65 +272,41 @@ String getEndpoint() {
...
-                assertEquals(expectedError.getCode(), actualError.getCode());
-                assertEquals(expectedError.getMessage(), actualError.getMessage());
+    private String createStorageAndGenerateSas(String folderPath) {
+        if (interceptorManager.isPlaybackMode()) {
+            return "https://isPlaybackmode";
+        } else {
+            String accountName = Configuration.getGlobalConfiguration().get("PRIMARY_STORAGE_ACCOUNT_NAME");
+            String accountKey = Configuration.getGlobalConfiguration().get("PRIMARY_STORAGE_ACCOUNT_KEY");
+            StorageSharedKeyCredential credential = new StorageSharedKeyCredential(accountName, accountKey);
+            String endpoint = String.format(Locale.ROOT, "https://%s.blob.core.windows.net", accountName);
```
Comment (danieljurek) [inline sdk/formrecognizer/azure-ai-formrecognizer/src/test/java/com/azure/ai/formrecognizer/FormTrainingClientTestBase.java:282]: `PRIMARY_STORAGE_ACCOUNT_NAME` should be something like `PRIMARY_STORAGE_ACCOUNT_URL` or this hardcoded URL (`https://%s.blob.core.windows.net`) will affect testing in other clouds.

Comment (samvaity) [reply sdk/formrecognizer/azure-ai-formrecognizer/src/test/java/com/azure/ai/formrecognizer/FormTrainingClientTestBase.java:282]: https://github.com/Azure/azure-sdk-for-java/issues/10380

File: sdk/formrecognizer/azure-ai-formrecognizer/CHANGELOG.md [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+## 1.0.0-beta.1 (Unreleased)
+Version 1.0.0-beta.1 is a preview of our efforts in creating a Azure Form Recognizer client library that is developer-friendly
+and idiomatic to the Java ecosystem. The principles that guide
+our efforts can be found in the [Azure SDK Design Guidelines for Java](https://azure.github.io/azure-sdk/java_introduction.html).
+
+For more information about this, and preview releases of other Azure SDK libraries, please visit
+https://azure.github.io/azure-sdk/releases/latest/java.html.
+
+- It uses the Form recognizer service `v2.0-preview.1` API.
```
Comment (mssfang) [inline sdk/formrecognizer/azure-ai-formrecognizer/CHANGELOG.md:11]: NIT:
```suggestion
- It uses the Form Recognizer service `v2.0-preview.1` API.
```

File: sdk/formrecognizer/azure-ai-formrecognizer/CHANGELOG.md [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+- Two client design:
+    - FormRecognizerClient to analyze fields/values on custom forms, receipts, and form content/layout
```
Comment (mssfang) [inline sdk/formrecognizer/azure-ai-formrecognizer/CHANGELOG.md:13]:
```suggestion
    - `FormRecognizerClient` to analyze fields/values on custom forms, receipts, and form content/layout
```

File: sdk/formrecognizer/azure-ai-formrecognizer/CHANGELOG.md [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+    - FormTrainingClient to train custom models (with/without labels), and manage the custom models on your account
```
Comment (mssfang) [inline sdk/formrecognizer/azure-ai-formrecognizer/CHANGELOG.md:14]:
```suggestion
    - `FormTrainingClient` to train custom models (with/without labels), and manage the custom models on your account
```

File: sdk/formrecognizer/azure-ai-formrecognizer/src/test/java/com/azure/ai/formrecognizer/TestUtils.java [hunk 8]
```diff
@@ -51,36 +51,35 @@
  * Contains helper methods for generating inputs for test methods
  */
 final class TestUtils {
-
-    static final String VALID_MODEL_ID = "95537f1b-aac4-4da8-8292-f1b93ac4c8f8";
-    static final String LABELED_MODEL_ID = "a0a3998a-b3c0-4075-aa6b-c4c4affe66b7";
     static final String INVALID_MODEL_ID = "a0a3998a-4c4affe66b7";
     static final String INVALID_STATUS_MODEL_ID = "22138c4e-c4b0-4901-a0e1-6c5beb73fc1d";
```
Comment (mssfang) [inline sdk/formrecognizer/azure-ai-formrecognizer/src/test/java/com/azure/ai/formrecognizer/TestUtils.java:55]: these two seems valid model id and status model id?

File: sdk/formrecognizer/azure-ai-formrecognizer/src/test/java/com/azure/ai/formrecognizer/TestUtils.java [hunk 9]
```diff
@@ -51,36 +51,35 @@
...
-    static final String VALID_UNLABELED_DATA_SAS_URL = "";
-    static final Object INVALID_MODEL_ID_ERROR = "Invalid UUID string: " + INVALID_MODEL_ID;
-    static final Object NULL_SOURCE_URL_ERROR = "'fileSourceUrl' cannot be null.";
+    static final String INVALID_RECEIPT_URL = "https://invalid.blob.core.windows.net/fr/contoso-allinone.jpg";
+    static final String INVALID_SOURCE_URL_ERROR = "Status code 400, \"{\"error\":{\"code\":\"1003\","
+        + "\"message\":\"Parameter 'Source' is not a valid Uri.\"}}\"";
+    static final String INVALID_MODEL_ID_ERROR = "Invalid UUID string: " + INVALID_MODEL_ID;
+    static final String NULL_SOURCE_URL_ERROR = "'fileSourceUrl' cannot be null.";
     static final String INVALID_URL = "htttttttps://localhost:8080";
+    static final String LABELED_MODEL_ID = "a0a3998a-b3c0-4075-aa6b-c4c4affe66b7";
```
Comment (mssfang) [inline sdk/formrecognizer/azure-ai-formrecognizer/src/test/java/com/azure/ai/formrecognizer/TestUtils.java:62]: valid model id and status model id?

File: sdk/formrecognizer/azure-ai-formrecognizer/src/test/java/com/azure/ai/formrecognizer/TestUtils.java [hunk 10]
```diff
@@ -172,7 +171,7 @@ static CustomFormModel getExpectedUnlabeledModel() {
             }
         };
         CustomFormSubModel customFormSubModel = new CustomFormSubModel(null, fieldMap, "form-0");
-        return new CustomFormModel(VALID_MODEL_ID, CustomFormModelStatus.READY,
+        return new CustomFormModel("95537f1b-aac4-4da8-8292-f1b93ac4c8f8", CustomFormModelStatus.READY,
```
Comment (mssfang) [inline sdk/formrecognizer/azure-ai-formrecognizer/src/test/java/com/azure/ai/formrecognizer/TestUtils.java:174]: create/use a\ static final variable for it.

File: sdk/formrecognizer/azure-ai-formrecognizer/src/test/java/com/azure/ai/formrecognizer/TestUtils.java (hunk 8 above)
Comment (samvaity) [reply sdk/formrecognizer/azure-ai-formrecognizer/src/test/java/com/azure/ai/formrecognizer/TestUtils.java:55]: >     static final String INVALID_MODEL_ID = "a0a3998a-4c4affe66b7";
It is not a valid UUID format for model Id.

>     static final String INVALID_STATUS_MODEL_ID = "22138c4e-c4b0-4901-a0e1-6c5beb73fc1d";
And this one has an invalid status.

File: sdk/formrecognizer/azure-ai-formrecognizer/src/test/java/com/azure/ai/formrecognizer/TestUtils.java (hunk 10 above)
Comment (samvaity) [reply sdk/formrecognizer/azure-ai-formrecognizer/src/test/java/com/azure/ai/formrecognizer/TestUtils.java:174]: It isn't reused anywhere else so didn't bother putting it in a static variable.

**Pull request merged**